rust/cloud-storage/email/src/domain/preview_metadata.rs

```diff
@@ -0,0 +1,61 @@
+use crate::domain::models::{
+    EmailThreadPreview, EmailThreadPreviewMetadata, IntermediateThreadMetadata,
+};
+use crate::domain::ports::EmailRepo;
+use crate::domain::service::EmailServiceImpl;
+use frecency::domain::ports::FrecencyQueryService;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+impl<T, U> EmailServiceImpl<T, U>
+where
+    T: EmailRepo,
+    U: FrecencyQueryService,
+{
+    /// get metadata for thread previews, used by FE for filtering Important vs Other threads
+    pub(crate) async fn get_preview_metadata(
```

Comment:
@seanaye in hexagaonal architecture shouldn't this be defined in the service namepsace and not the domain namespace?